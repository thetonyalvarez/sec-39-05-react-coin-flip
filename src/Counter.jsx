import React, { useState } from 'react'
import Grid from '@mui/material/Grid'
import Card from '@mui/material/Card'
import Typography from '@mui/material/Typography'

const Counter = (props) => {

  return (
    <Grid
      container
      justifyContent="center"
      spacing={ 2 }
      sx={{my: 4}}>

      <Grid item xs={12} sm={12} md={12} lg={3}>
        <Card sx={{p: 4}}>
          <Typography variant="subtitle2">Total Flips</Typography>
          <Typography variant="h3">{props.total}</Typography>
        </Card>
      </Grid>
      <Grid item xs={12} sm={6} md={6} lg={3}>
        <Card sx={{p: 4}}>
            <Typography variant="subtitle2">Heads</Typography>
            <Typography variant="h3">{props.heads}</Typography>
          </Card>
      </Grid>
      <Grid item xs={12} sm={6} md={6} lg={3}>
        <Card sx={{p: 4}}>
            <Typography variant="subtitle2">Tails</Typography>
            <Typography variant="h3">{props.tails}</Typography>
          </Card>
      </Grid>
    </Grid>
  )
}

export default Counter