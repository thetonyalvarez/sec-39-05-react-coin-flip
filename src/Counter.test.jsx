import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { expect, it, describe } from 'vitest'
import Counter from "./Counter";

it("renders without crashing", function() {
  render(<Counter/>)
});

it("matches snapshot", function() {
  const {asFragment} = render(<Counter />);
  expect(asFragment()).toMatchSnapshot();
});