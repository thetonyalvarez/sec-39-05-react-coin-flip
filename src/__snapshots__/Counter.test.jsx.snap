// Vitest Snapshot v1

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-5z7jm9-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-3 css-1ozs4dl-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-xt7fqp-MuiPaper-root-MuiCard-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
        >
          Total Flips
        </h6>
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-6 MuiGrid-grid-lg-3 css-1eehwzh-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-xt7fqp-MuiPaper-root-MuiCard-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
        >
          Heads
        </h6>
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-6 MuiGrid-grid-lg-3 css-1eehwzh-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-xt7fqp-MuiPaper-root-MuiCard-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
        >
          Tails
        </h6>
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
